statsmodels/tsa/statespace/sarimax.py

```diff
@@ -0,0 +1,1941 @@
+        maximum likelihood estimation or through the Kalman filter (i.e.
+        recursive least squares). If `time_varying_regression` is True, this
+        must be set to False. Default is True.
+    simple_differencing : boolean, optional
+        Whether or not to use conditional maximum likelihood estimation.
+        If True, differencing is performed prior to estimation, which discards
+        the first :math:`SD + d` initial rows but reuslts in a smaller
+        state-space formulation. If False, the full SARIMAX model is put in
+        state-space form so that all datapoints can be used in estimation.
+        Default is False.
+    enforce_stationarity : boolean, optional
+        Whether or not to transform the AR parameters to enforce stationarity
+        in the autoregressive component of the model. Default is True.
+    enforce_invertibility : boolean, optional
+        Whether or not to transform the MA parameters to enforce invertibility
+        in the moving average component of the model. Default is True.
+    hamilton_representation : boolean, optional
+        Whether or not to use the Hamilton representation of an ARMA process
+        (if True) or the Harvey representation (if False). Default is False.
+
+    Attributes
+    ----------
+    measurement_error : boolean
+        Whether or not to assume the observations were measured with error.
+    state_error : boolean
+        Whether or not the transition equation has an error component.
+    mle_regression : boolean
+        Whether or not to use estimate the regression coefficients as part of
+        maximum likelihood estimation.
+    state_regression : boolean
+        Whether or not to use estimate the regression coefficients via the
+        Kalman filter, by extending the state space.
+    time_varying_regression : boolean
+        Whether or not coefficients on the exogenous regressors are allowed to
+        vary over time. Not yet implemented.
+    simple_differencing : boolean
+        Whether or not to use conditional maximum likelihood estimation.
+    enforce_stationarity : boolean
+        Whether or not to transform the AR parameters to enforce stationarity
+        in the autoregressive component of the model.
+    enforce_invertibility : boolean
+        Whether or not to transform the MA parameters to enforce invertibility
+        in the moving average component of the model.
+    hamilton_representation : boolean
+        Whether or not to use the Hamilton representation of an ARMA process.
+    trend : str{'n','c','t','ct'} or iterable
+        Parameter controlling the deterministic trend polynomial :math:`A(t)`.
+        See the class parameter documentation for more information.
+    polynomial_ar : array
+        Array containing autoregressive lag polynomial coefficients,
+        ordered from lowest degree to highest. Initialized with ones, unless
+        a coefficient is constrained to be zero (in which case it is zero).
```

Comment:
Maybe they are not attached to ARIMA. I used the full polynomials in the ARMAProcess class, since almost all calculations are easier as polynomial.

the docstring explanation maybe just drop "coefficients" if you include the leading 1.
in ARMAProcess I used `ar_coeffs` `ma_coeffs`  for just the coefficients (excluding the ones) with the sign for the RHS.
maybe possible rename `ar_poly` and `ma_poly` but I don't know if there is already a precedence case if ARIMA doesn't have it.